include/yaclib/async/when_all.hpp

```diff
@@ -21,16 +22,17 @@ namespace yaclib {
  * \param begin , count the range of futures to combine
  * \return Future<std::vector<future_value_t<T>>, future_error_t<T>>
  */
-template <WhenPolicy P = WhenPolicy::FirstFail, typename It, typename T = typename std::iterator_traits<It>::value_type>
+template <FailPolicy F = FailPolicy::FirstFail, OrderPolicy O = OrderPolicy::Fifo, typename It,
+          typename T = typename std::iterator_traits<It>::value_type>
 auto WhenAll(It begin, std::size_t count) {
-  static_assert(P != WhenPolicy::LastFail, "TODO(Ri7ay, MBkkt) Add other policy for WhenAll");
+  static_assert(F != FailPolicy::LastFail, "TODO(Ri7ay, MBkkt) Add other policy for WhenAll");
   static_assert(is_future_base_v<T>, "WhenAll function Iterator must be point to some Future");
   YACLIB_WARN(count < 2, "Don't use combinators for one or zero futures");
   using V = future_base_value_t<T>;
   using E = future_base_error_t<T>;
-  using R = std::conditional_t<P == WhenPolicy::None, Result<V, E>, V>;
-  auto [future_core, combinator] = detail::AllCombinator<R, E>::Make(count);
-  detail::WhenImpl(combinator, begin, count);
+  using R = std::conditional_t<F == FailPolicy::None, Result<V, E>, V>;
```

Comment:
In general no difference between Fifo and Same if R is void.
So I think it always should be Fifo policy, and Same AllCombinator shouldn't work with void